src/renderer/atlas/CustomGlyphs.cpp

```diff
@@ -0,0 +1,1265 @@
+    },
+    // U+259A ▚ QUADRANT UPPER LEFT AND LOWER RIGHT
+    {
+        Instruction{ Shape_Filled100, Pos_0_1, Pos_0_1, Pos_1_2, Pos_1_2 },
+        Instruction{ Shape_Filled100, Pos_1_2, Pos_1_2, Pos_1_1, Pos_1_1 },
+    },
+    // U+259B ▛ QUADRANT UPPER LEFT AND UPPER RIGHT AND LOWER LEFT
+    {
+        Instruction{ Shape_Filled100, Pos_0_1, Pos_0_1, Pos_1_2, Pos_1_1 },
+        Instruction{ Shape_Filled100, Pos_1_2, Pos_0_1, Pos_1_1, Pos_1_2 },
+    },
+    // U+259C ▜ QUADRANT UPPER LEFT AND UPPER RIGHT AND LOWER RIGHT
+    {
+        Instruction{ Shape_Filled100, Pos_0_1, Pos_0_1, Pos_1_2, Pos_1_2 },
+        Instruction{ Shape_Filled100, Pos_1_2, Pos_0_1, Pos_1_1, Pos_1_1 },
+    },
+    // U+259D ▝ QUADRANT UPPER RIGHT
+    {
+        Instruction{ Shape_Filled100, Pos_1_2, Pos_0_1, Pos_1_1, Pos_1_2 },
+    },
+    // U+259E ▞ QUADRANT UPPER RIGHT AND LOWER LEFT
+    {
+        Instruction{ Shape_Filled100, Pos_0_1, Pos_1_2, Pos_1_2, Pos_1_1 },
+        Instruction{ Shape_Filled100, Pos_1_2, Pos_0_1, Pos_1_1, Pos_1_2 },
+    },
+    // U+259F ▟ QUADRANT UPPER RIGHT AND LOWER LEFT AND LOWER RIGHT
+    {
+        Instruction{ Shape_Filled100, Pos_0_1, Pos_1_2, Pos_1_2, Pos_1_1 },
+        Instruction{ Shape_Filled100, Pos_1_2, Pos_0_1, Pos_1_1, Pos_1_1 },
+    },
+};
+
+static constexpr char32_t Powerline_FirstChar = 0xE0B0;
+static constexpr u32 Powerline_CharCount = 0x10;
+static constexpr Instruction Powerline[Powerline_CharCount][InstructionsPerGlyph] = {
+    // U+E0B0 Right triangle solid
+    {
+        Instruction{ Shape_ClosedFilledPath, Pos_0_1, Pos_0_1, Pos_1_1, Pos_1_2 },
+        Instruction{ Shape_ClosedFilledPath, Pos_0_1, Pos_1_1, Pos_Min, Pos_Min },
+    },
+    // U+E0B1 Right triangle line
+    {
+        Instruction{ Shape_OpenLinePath, Pos_0_1, Pos_0_1, Pos_1_1_Sub_0_5, Pos_1_2 },
+        Instruction{ Shape_OpenLinePath, Pos_0_1, Pos_1_1, Pos_Min, Pos_Min },
+    },
+    // U+E0B2 Left triangle solid
+    {
+        Instruction{ Shape_ClosedFilledPath, Pos_1_1, Pos_0_1, Pos_0_1, Pos_1_2 },
+        Instruction{ Shape_ClosedFilledPath, Pos_1_1, Pos_1_1, Pos_Min, Pos_Min },
+    },
+    // U+E0B3 Left triangle line
+    {
+        Instruction{ Shape_OpenLinePath, Pos_1_1, Pos_0_1, Pos_0_1_Add_0_5, Pos_1_2 },
+        Instruction{ Shape_OpenLinePath, Pos_1_1, Pos_1_1, Pos_Min, Pos_Min },
+    },
+    // U+E0B4 Right semi-circle solid
+    {
+        Instruction{ Shape_FilledEllipsis, Pos_0_1, Pos_1_2, Pos_1_1, Pos_1_2 },
+    },
+    // U+E0B5 Right semi-circle line
+    {
+        Instruction{ Shape_EmptyEllipsis, Pos_0_1, Pos_1_2, Pos_1_1_Sub_0_5, Pos_1_2_Sub_0_5 },
+    },
+    // U+E0B6 Left semi-circle solid
+    {
+        Instruction{ Shape_FilledEllipsis, Pos_1_1, Pos_1_2, Pos_1_1, Pos_1_2 },
+    },
+    // U+E0B7 Left semi-circle line
+    {
+        Instruction{ Shape_EmptyEllipsis, Pos_1_1, Pos_1_2, Pos_1_1_Sub_0_5, Pos_1_2_Sub_0_5 },
+    },
```

Comment:
Looking at these codes

```cpp

        auto endX = Pos_Lut[instruction.endX][0] * rectW + rectX;
        auto endY = Pos_Lut[instruction.endY][0] * rectH + rectY;

// later
case Shape_FilledEllipsis:
        case Shape_EmptyEllipsis:
        {
            const D2D1_ELLIPSE e{ { begX, begY }, endX, endY };
            if (shape == Shape_FilledEllipsis)
            {
                renderTarget->FillEllipse(&e, brush);
            }
            else
            {
                renderTarget->DrawEllipse(&e, brush, lineWidth, nullptr);
            }
            break;
        }
```

I think `endX`, `endY` should not be added rectX and rectY because they are radius?

```cpp
const D2D1_ELLIPSE e{ { begX, begY }, endX - rectX, endY - rectY };
```

Reply:
Yeah you're right. I'll change it up a bit so that the rectX/rectY is added as late as possible.